Sources/Controllers/2.0/OACustomPOIViewController.mm

```diff
@@ -188,135 +156,109 @@ - (void)selectSubcategoryCancel
+        cell.separatorInset = UIEdgeInsetsMake(0.0, 66.0, 0.0, 0.0);
+
+        cell.textView.text = item.nameLocalized;
+        cell.textView.textColor = [UIColor blackColor];
+
+        cell.overflowButton.enabled = NO;
+        cell.overflowButton.imageView.contentMode = UIViewContentModeCenter;
+        [cell.overflowButton setImage:[[UIImage imageNamed:@"ic_custom_arrow_right"] imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate] forState:UIControlStateNormal];
+        cell.overflowButton.tintColor = UIColorFromRGB(color_tint_gray);
+
+        UIImage *img = [[item icon] imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate];
+        cell.iconView.image = img;
+        cell.iconView.tintColor = isSelected ? UIColorFromRGB(color_primary_purple) : UIColorFromRGB(color_tint_gray);
+
+        NSString *descText;
+        if (subtypes == [OAPOIBaseType nullSet] || countAllTypes == countAcceptedTypes)
+            descText = [NSString stringWithFormat:@"%@ - %lu", OALocalizedString(@"shared_string_all"), countAllTypes];
         else
-        {
-            UIImage *img = [[item icon] imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate];
-            [cell.imgView setImage: img];
-            cell.descriptionView.hidden = YES;
-            cell.switchView.on = NO;
-        }
+            descText = [NSString stringWithFormat:@"%lu/%lu", countAcceptedTypes, countAllTypes];
+        cell.descView.text = descText;
+        cell.descView.textColor = UIColorFromRGB(color_text_footer);
+
         [cell updateConstraintsIfNeeded];
     }
     return cell;
 }
 
+- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
+{
+    [tableView deselectRowAtIndexPath:indexPath animated:YES];
+    
+    OAPOICategory* item = _categories[indexPath.row];
+    OASelectSubcategoryViewController *subcategoryScreen = [[OASelectSubcategoryViewController alloc] initWithCategory:item filter:_filter];
+    subcategoryScreen.delegate = self;
+    subcategoryScreen.modalPresentationStyle = UIModalPresentationFullScreen;
+    [self showViewController:subcategoryScreen];
+}
 
 - (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
 {
-    return _dataArray.count;
+    return _categories.count;
 }
 
--(void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
+- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath;
 {
-    [tableView deselectRowAtIndexPath:indexPath animated:YES];
-    
-    OAPOICategory* item = _dataArray[indexPath.row];
-    OASelectSubcategoryViewController *subcatController = [[OASelectSubcategoryViewController alloc] initWithCategory:item subcategories:[_filter getAcceptedSubtypes:item] selectAll:[_filter getAcceptedSubtypes:item] == [OAPOIBaseType nullSet]];
-    subcatController.delegate = self;
-    [self.navigationController pushViewController:subcatController animated:YES];
+    return 66.0;
```

Comment:
Do not hardcode heightForRowAtIndexPath, set AutomaticDimensions in xib and remove this method completely.